pkg/config/install_requirements.go

```diff
@@ -520,10 +520,10 @@ func LoadRequirementsConfig(dir string) (*RequirementsConfig, string, error) {
 	for absolute != "" && absolute != "." && absolute != "/" {
 		fileName := filepath.Join(absolute, RequirementsConfigFileName)
 		config, err := LoadRequirementsConfigFile(fileName)
-		if err == nil {
-			return config, fileName, nil
+		if err != nil {
+			return nil, "", err
```

Comment:
I don't think this is quite correct. If you look at the docs for this function, then you see that the intend is to walk up the filesystem to find the requirements file there. I am not saying this is good behaviour, just how it is. The reason for this is to be able to execute individual jx boot step in sub-directories of the requirements/dev repository. 

I am pretty sure that this change will brake this behaviour. 

To do what you are after, I think we need to split or at least pull out the file existence check from `LoadRequirementsConfigFile`. You need to be able to distinguish between the file not existing at all and the file not loadable due to a syntax error.

I think it would be great to add some tests around this as well. Should be quite straight forward, since _install_requirements_test.go_ already exists. One can just add a couple of more test scenarios to the _test_data_ directory. 

WDYT?

Reply:
Thank you for your review! I know how to fix it.

Reply:
Resolved it.

Reply:
I still would feel more comfortable if we prior to `config, err := LoadRequirementsConfigFile(fileName)` do an explicit file existence check:

```go
        absolute = filepath.Dir(absolute)
	exists, err := util.FileExists(fileName)
	if err != nil {
		return nil, "", err
	}

        if !exist {
            continue
       }

       config, err := LoadRequirementsConfigFile(fileName)
	if err == nil {
		return nil, "", err
	}
```

Something along these lines. I think that is easier to understand and it does also not rely on an implementation detail of `LoadRequirementsConfigFile`, namely that it does return the PathError directly. It might as well wrap it.

Reply:
I feel great!